### PR TITLE
Split Webhook into type specific definitions

### LINKED ISSUE #608: Split up webhook into 3 different models
### Summary
This should bring in support for the new webhook type in https://github.com/discord/discord-api-docs/pull/2926 along with splitting the webhook class into 3 different types in a similar fashion to how channels are handled. In this new structure only IncomingWebhooks should be executable (only incoming webhooks and possibly following webhooks should be mutable rest wise).

This will likely be blocked until #577 is merged due to the changes that already makes to webhooks. 

### Problem
Interaction webhooks aren't supported and support should be added without muddying up the semantics of any existing class. 

### Ideal implementation
See summary

### Checklist

- [x] I have searched the issue tracker and have made sure it's not a duplicate.
      If it is a follow up of another issue, I have specified it.

### PR DESCRIPTION
**Pull request by FasterSpeeding**
### Summary
Split Webhook into type specific definitions

### Checklist

- [x] I have run `nox` and all the pipelines have passed.
- [x] I have made unittests according to the code I have added/modified/deleted.

### Related issues
Closes #608

### CODE REVIEW

File: hikari/webhooks.py [hunk 1]
```diff
@@ -761,71 +798,164 @@ async def fetch_self(self, *, use_token: undefined.UndefinedOr[bool] = undefined
...
 
-        return routes.CDN_USER_AVATAR.compile_to_file(
-            urls.CDN_URL,
-            user_id=self.id,
-            hash=self.avatar_hash,
-            size=size,
-            file_format=ext,
-        )
+
+@attr.define(hash=True, kw_only=True, weakref_slot=False)
```
Comment (davfsa) [inline hikari/webhooks.py:953]:
```suggestion
@attr.define(hash=True, kw_only=True, weakref_slot=False, auto_attribs=False)
```

File: hikari/webhooks.py [hunk 2]
```diff
@@ -59,6 +66,9 @@ class WebhookType(int, enums.Enum):
     CHANNEL_FOLLOWER = 2
     """Channel Follower webhook."""
 
+    APPLICATION = 3
+    """Application webhook (from the interactions flow)."""
```
[the remark this answers is not in the document]
Comment (FasterSpeeding) [reply hikari/webhooks.py:70]: That still is and always has been an incoming webhook

File: hikari/webhooks.py (hunk 1 above)
Comment (FasterSpeeding) [reply hikari/webhooks.py:953]: Don't see why that'd need to be disabled, doesn't really change anything here

Comment (davfsa) [reply hikari/webhooks.py:953]: Stops attr from adding attributes to it, doesnt hurt to add

Comment (FasterSpeeding) [reply hikari/webhooks.py:953]: Gonna keep redefining the attr field since this ensures field.type is correct for this field

**Review by davfsa — APPROVED** [review]

**Pull request merged**